### DISCUSSION TITLE
Secondary level includes

### DISCUSSION THREAD

**Discussion by etiennecl** [JasperFx/marten#2116] [Ideas]
I've looked in the documentation thoroughly and I haven't seen the ability to query documents using secondary level includes. 

We would like the ability to declare secondary level includes or event just includes via a String-based includes functionality:

```cs
query.Query<Issue>().Include("User.Profile");
```

**Comment by jeremydmiller** [comment]
I'm gonna ixnay the string based includes because of the potential bugginess, but you can already do `.Include(x => x.User.Profile)` I believe.

**Comment by etiennecl** [comment]
I know it's not ideal as a developer could very easily mistype the properties inside the include, but I still think it should be considered to provide some more dynamic query builder capabilities.

I have a use case where I would of like marten to extend on its `Schema Feature Extensions` capabilities to be able to define new relationships at runtime (this is already provided via the Table class). But then there is currently no way to dynamically query relationships that have been defined at runtime. This is why I think string-based include could potentially "solve" my use case. 

I'm aware that it's an advanced use case and that some of the capabilities would increase the potential of bugs, but I still think marten should provide some low-level API for those kinds of use cases.

*I really appreciate all the work and features that this library is offering.
